src/Map.ts

```diff
@@ -0,0 +1,261 @@
+import { hasClass } from './decorators';
+import {is} from './object';
+import global from './global';
+
+module Native {
```

Comment:
At this point, Forbes and I are thinking that it may be more prudent for core's `Map` module to settle for implementing a "Map-like" interface (what the `Shim` currently does) rather than trying to wrap native if it exists.  In particular, this will avoid conundrums around variance in platforms' support of Iterable objects.

So we should be able to remove `module Native` entirely.